Kernel/Syscalls/process.cpp

```diff
@@ -64,4 +64,17 @@ int Process::sys$set_process_name(Userspace<const char*> user_name, size_t user_
     return 0;
 }
 
+// FIXME: If we can get a syscall function with 5 arguments, we could pass key/value here.
```

Comment:
Not sure what to do about this. Sending a single "key:value" string is probably an ok-ish solution, although the internet suggests syscalls with more than three arguments should be possible...

Reply:
what's wrong with the existing way? just put all five into a struct and pass the struct's address to the kernel?

Reply:
🤯

You see, I have clearly no idea what I'm doing.

---

So, something like this? (names need some work, obvs)

```cpp
struct KeyAndValue {
    Userspace<const char*> key;
    size_t key_length;
    Userspace<const char*> value;
    size_t value_length;
};

int Process::sys$set_coredump_metadata(Userspace<const KeyAndValue*> key_and_value)
{
}
```

Reply:
Right, look at `sys$mmap` for example. All the `SC_foo_params` structs are defined in `Kernel/API/Syscall.h`

Reply:
Done, amazing!